### PR TITLE
A branch for experiments

### PR DESCRIPTION
**Pull request by RPC2**

### CODE REVIEW

File: src/il_representations/scripts/chain_configs.py [hunk 1]
```diff
@@ -0,0 +1,507 @@
+from il_representations.scripts.utils import StagesToRun
```
Comment (qxcv) [inline src/il_representations/scripts/chain_configs.py:1]: This file seems to be a duplicate of [`configs/chain_configs.py`](https://github.com/HumanCompatibleAI/eirli/blob/cyn-expt/src/il_representations/configs/chain_configs.py). Was it accidentally included in this PR?

File: src/il_representations/scripts/hp_tuning.py [hunk 2]
```diff
@@ -0,0 +1,299 @@
+from il_representations.algos import augmenters, pair_constructors, encoders, losses, batch_extenders
```
Comment (qxcv) [inline src/il_representations/scripts/hp_tuning.py:1]: Same comment on this file as on `chain_configs.py`: it seems to be a duplicate of another file in the `configs/` directory. I assume this one is old and we should remove it?

File: src/il_representations/configs/chain_configs.py [hunk 3]
```diff
@@ -418,11 +418,29 @@ def cfg_repl_temporal_cpc():
         stages_to_run = StagesToRun.REPL_AND_IL
         repl = {
             'algo': 'TemporalCPC',
+
```
Comment (qxcv) [inline src/il_representations/configs/chain_configs.py:421]:
```suggestion
```

Deleting extra space.

File: src/il_representations/configs/chain_configs.py [hunk 4]
```diff
@@ -418,11 +418,29 @@ def cfg_repl_temporal_cpc():
...
+    def cfg_repl_tcpc8():
+        stages_to_run = StagesToRun.REPL_AND_IL
+        repl = {
+            'algo': 'TemporalCPC',
+            'algo_params': {
+                'target_pair_constructor_kwargs': {
+                    'temporal_offset': 8
+                }
+            }
+
```
Comment (qxcv) [inline src/il_representations/configs/chain_configs.py:437]:
```suggestion
```

Same extra space issue here.

File: src/il_representations/scripts/interpret.py [hunk 5]
```diff
@@ -208,161 +136,71 @@ def deep_lift(net, tensor_image, label):
...
+    def __init__(self, policy, use_first_action):
+        super().__init__()
+        self.policy = policy
+        self.use_first_action = use_first_action
 
-    if module_name == 'encoder':
-        rep_encoder = network.policy.features_extractor.representation_encoder
+    def forward(self, x):
+        # TODO: make this compatible with both DMC and procgen.
+        action = self.policy.action_net(self.policy.features_extractor(x/255.))
```
Comment (qxcv) [inline src/il_representations/scripts/interpret.py:178]: Maybe add a comment to clarify what `action_net` produces? Originally I assumed it would produce an action (which is a problem for procgen, since you can't backpropagate through the discrete action sampling mechanism). On reading the SB3 source, though, it seems like it either produces (1) logits for a categorical distribution, or (2) a mean vector for a Gaussian distribution. Writing this on the line above would be useful to future readers.

File: src/il_representations/scripts/interpret.py [hunk 6]
```diff
@@ -371,42 +209,44 @@ def run(log_dir, chosen_algo, layer_kwargs, save_video, filename, dataset_config
...
+            save_img(save_name=f'{filename}_{itr}',
+                     save_dir=f'{log_dir}/images')
+
+            if save_original_image:
+                processed_image = []
+                i = 0
+                while i < len(image):
+                    processed_image.append(1.0/3.0 * image[i:i+3].astype(float))
+                    i += 3
+                processed_image = np.sum(processed_image, axis=0).astype(int)
```
Comment (qxcv) [inline src/il_representations/scripts/interpret.py:243]:
```suggestion
                # convert to grayscale
                processed_image = np.mean(
                    image.reshape((3, -1) + image.shape[1:]), axis=0).astype(int)
```

I think (need to check) that this does the same thing as the original code, and it's a bit shorter/more straightforward.

Comment (qxcv) [reply src/il_representations/scripts/interpret.py:243]: Also, meta-level point: I assumed this was converting to grayscale and discarding the channels axis, but then I'm a bit confused by the transpose line down the bottom, which seems to assume that axis 0 is still the channels axis. Could you please add a comment explaining what the data format of `image` is, and why this transformation makes sense?

**Review by qxcv — CHANGES_REQUESTED** [review]
A few small comments. Main one is that there seem to be two duplicate files in `src/il_representations/scripts/{chain_config,hp_tuning}.py`. Are those meant to be there? They seem to be old copies, since the config files got moved to ``src/il_representations/configs` (rather than ``src/il_representations/scripts`) in an earlier PR.

File: src/il_representations/scripts/hp_tuning.py (hunk 2 above)
Comment (RPC2) [reply src/il_representations/scripts/hp_tuning.py]: Yes! These two files are mistakenly kept under `scripts`. I have already removed it.

File: src/il_representations/scripts/interpret.py (hunk 6 above)
Comment (qxcv) [reply src/il_representations/scripts/interpret.py]: Okay, with the comment it makes sense. It's just averaging the three input frames right?

Comment (RPC2) [reply src/il_representations/scripts/interpret.py]: Yes :)
